### ISSUE TITLE
Old-Style AsciiDoc Headers

### ISSUE THREAD

**Issue by znerd** [jgm/pandoc#5038]
Pandoc 2.3.1 outputs old-style AsciiDoc headers instead of new-style headers.

**Steps to reproduce:**
1. Have a simple HTML file in your working directory, such as this one:
```
<html>
<head>
<title>Document Title</title>
</head>
<body>
<h1>Level 1</h1>
<p>Text after level 1 heading.
<h2>Level 2</h2>
<p>Text after level 2 heading.
<h3>Level 3</h3>
<p>Text after level 3 heading.
</body>
</html>
```
2. Let Pandoc convert from HTML to AsciiDoc:
```
pandoc -f html -t asciidoc -o output.adoc input.html
```
3. Check the generated AsciiDoc file (`output.adoc`).

**Expected results:**
1. L1 header is using new-style AsciiDoc header syntax:
```
== Level 1
```
2. L2 header is using new-style AsciiDoc header syntax:
```
=== Level 2
```
3. L3 header is using new-style AsciiDoc header syntax:
```
==== Level 3
```

**Actual results:**
1. NOK: L1 header is using old-style AsciiDoc header syntax:
```
Level 1
-------
```
2. NOK: L2 header is using old-style AsciiDoc header syntax:
```
Level 2
~~~~~~~
```
3. NOK L3 header is using old-style AsciiDoc header syntax:
```
Level 3
^^^^^^^
```

References:
- https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#section-titles

**Comment by mb21** [comment]
AFAIK this is intended. The title is considered metadata, not part of the document body text. E.g. for markdown the equivalent would be part of the YAML metadata. Not sure there is an asciidoc equivalent.

**Comment by jgm** [comment]
Correct, the title won't appear unless you use `-s`; the way it appears is controlled by the default asciidoc template.  Please use `-s` to see how this works.  It's possible that the asciidoc template also needs updating to use the new section header syntax.

**Comment by jgm** [comment]
But wait: you're linking to asciidoctor documentation.  Is this change in header syntax part of official asciidoc?  Not as far as I can see.

We try to conform to the official asciidoc syntax.

**Comment by znerd** [comment]
The AsciiDoc format/standard and toolchain (see [asciidoc.org](http://asciidoc.org/) and [asciidoc/asciidoc](https://github.com/asciidoc/asciidoc) on GitHub) was actively maintained until 2013.

Then since 2013/2014 the stick has been taken over by Asciidoctor (see [asciidoctor.org](https://asciidoctor.org/)). Evidence: the [README.asciidoc](https://github.com/asciidoc/asciidoc/blob/master/README.asciidoc) file on the AsciiDoc project in GitHub.

According to the AsciiDoc site, there are 2 supported types of section headings: single line, and two-line headings; evidence: [AsciiDoc User Guide, chapter 11 “Titles”](http://asciidoc.org/userguide.html#X17).

On the Asciidoctor site, while implicit, it is obvious that the two-line headings are preferred over single-line headings. Evidence:
- The [Asciidoctor home page](https://asciidoctor.org/) shows single-line headings.
- The [Asciidoctor Writer's Guide](https://asciidoctor.org/docs/asciidoc-writers-guide/) does not even mention the two-line form, while it does explicitly address headings, see the section [Section Titles](https://asciidoctor.org/docs/asciidoc-writers-guide/#section-titles).

**Comment by znerd** [comment]
Note also that the AsciiDoc plugin for IntelliJ IDEA warns about two-line headings. It calls them “old-style headings.”

Evidence: [`AsciidocHeadingStyleInspection.java`](https://github.com/asciidoctor/asciidoctor-intellij-plugin/blob/master/src/main/java/org/asciidoc/intellij/inspections/AsciidocHeadingStyleInspection.java) (in GitHub project `asciidoctor/asciidoctor-intellij-plugin`).

**Comment by jgm** [comment]
OK, thanks for clearing that up!

**Comment by znerd** [comment]
@jgm : Thanks for the quick fix; commit looks good!

**Comment by mb21** [comment]
Should we update the asciidoc [link in the manual](https://github.com/jgm/pandoc/blob/master/MANUAL.txt#L451) to https://asciidoctor.org then?

**Comment by znerd** [comment]
@mb21 I'm not convinced that's the right thing to do. While AsciiDoctor is the primarily vehicle for the format and toolchain, the AsciiDoc ecosystem is bigger than just Asciidoctor.

But perhaps it does make sense to update the documentation somehow, to reflect that Asciidoctor now seems to be carrying forth the torch.
